### PR TITLE
feat: pytorch math-operations-a

### PR DESCRIPTION
**Pull request by Breeze-P**
在compat_pytorch中增加了对math-operations的支持，并添加了相关的单元测试。

### CODE REVIEW

File: brainpy/_src/math/compat_pytorch.py [hunk 1]
```diff
@@ -86,3 +86,145 @@ def unsqueeze(input: Union[jax.Array, Array], dim: int) -> Array:
...
+    return _return(r)
+  else:
+    _check_out(out)
+    out.value = r
+
+arccosh = acosh
+
+def add(input: Union[jax.Array, Array, jnp.number],
+        other: Union[jax.Array, Array, jnp.number],
+        *, alpha: Optional[jnp.number] = 1,
```
Comment (chaoming0625) [inline brainpy/_src/math/compat_pytorch.py:130]: where is `alpha` used?

**Review by chaoming0625 — CHANGES_REQUESTED** [review]
A good PR. Minor bugs to fix:

1. `alpha` is redundant in `add()` function?
2. Many functions have incorrect return typing. They should be `-> Optional[Array]` rather than `Array`.

**Comment by chaoming0625** [comment]
One more thing, add these functions in ``brainpy.math.compat_torch.py``.

File: brainpy/_src/math/compat_pytorch.py (hunk 1 above)
Comment (Breeze-P) [reply brainpy/_src/math/compat_pytorch.py:130]: In Pytorch, the `add` func is not only "add". It also means "Adds other, scaled by alpha, to input.", and `alpha` is used in line 134.

**Review by chaoming0625 — APPROVED** [review]
Excellent~

**Pull request merged**